File: problem_solving_for_beginners/string_problems/python/palindrome.py
#!/usr/bin/python3
def length(string):
    """Length of a string"""
    count = 0
    for i in string:
        count += 1
    return count

def palindrome(string):
    """Copy string"""
    copy = str1[:]
    """Reverse string"""
    rev_str = str1[::-1]
    """Compare copied and reversed string"""
    if (length(copy) == length(rev_str)):
        for i in range(length(str1)):
            if rev_str[i] != copy[i]:
                print("Not Palindrome")
                break
        else:
            print("Palindrome")
    else:
        print("Not Palindrome")

def palindrome2(string):
    flag = 1
    strlen = length(string)
    lastidx = strlen - 1
    for i in range(strlen//2):
        if string[i] != string[lastidx - i]:
            flag = 0
            print("Non Palindrome")
            break
        else:
            print("Palindrome")

if __name__ == "__main__":
    str1 = input("Enter a string: ")
    # palindrome(str1)
    palindrome2(str1)

File: problem_solving_for_beginners/string_problems/c-programming/palindrome.c
#include <stdio.h>
#include <stdlib.h>
/**
 * palindrome - Check wheather a string is a palindrome or not
 * @str: String to check
 * Return: print Palindrome Otherwise Not Palindrome
 */
void palindrome(char *str)
{
    char *copy, tmp;
    int len = 0, i, lastidx;

    // find the length
    for (i = 0; str[i]; i++)
        len++;

    // Allocate memory
    copy = malloc(sizeof(char) * len);

    // copy the string
    for (i = 0; str[i]; i++)
        copy[i] = str[i];
    copy[i] = '\0';

    // Reverse the original string
    lastidx = len - 1;
    for (i = 0; i < len/2; i++)
    {
        tmp = str[i];
        str[i] = str[lastidx - i];
        str[lastidx - i] = tmp;
    }

    // Compare the reversed string with copied string
    i = 0;
    while (str[i] != '\0' && copy[i] != '\0' && str[i] == copy[i])
        i++;

    // print Palindrome if they are eqaul otherwise Non Palindrome
    if (str[i] == '\0' && copy[i] == '\0')
        printf("Palindrome\n");
    else
        printf("Not Palindrome\n");

    free(copy);
    copy = NULL;
}

int main(void)
{
	char str[10];
	printf("Enter a string: ");
	scanf("%s", str);

	//Write your code here
	palindrome(str);
	return 0;
}